modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/CollectorRegistryPrometheus.scala

```diff
@@ -41,40 +63,68 @@ object CollectorRegistryPrometheus {
 
     new CollectorRegistry[F] {
 
-      def gauge[A, B[_]](
+      def gauge[A, B[_], D[_]](
         name: String,
         help: String,
         labels: A)(implicit
-        magnet: LabelsMagnet[A, B]
+        magnet: LabelsMagnet[A, B, D]
       ) = {
         val gauge = P.Gauge.build()
           .name(name)
           .help(help)
 
-        apply[A, B, P.Gauge.Child, P.Gauge, P.Gauge.Builder, Gauge[F]](gauge, labels)
+        apply[A, B, D, P.Gauge.Child, P.Gauge, P.Gauge.Builder, Gauge[F]](gauge, labels, List.empty)
       }
 
+      def gaugeWithInitialLabels[A, B[_], D[_]](
+        name: String,
+        help: String,
+        labels: A)(implicit
+        magnet: LabelsMagnet[A, B, D]
+      ) = {
+        magnet.withInitialLabelValues { initialLabelValues =>
+          val gauge = P.Gauge.build()
+            .name(name)
+            .help(help)
+
+          apply[A, B, D, P.Gauge.Child, P.Gauge, P.Gauge.Builder, Gauge[F]](gauge, labels, initialLabelValues)
+        }
+      }
 
-      def counter[A, B[_]](
+      def counter[A, B[_], D[_]](
         name: String,
         help: String,
         labels: A)(implicit
-        magnet: LabelsMagnet[A, B]
+        magnet: LabelsMagnet[A, B, D]
       ) = {
         val counter = P.Counter.build()
           .name(name)
           .help(help)
 
-        apply[A, B, P.Counter.Child, P.Counter, P.Counter.Builder, Counter[F]](counter, labels)
+        apply[A, B, D, P.Counter.Child, P.Counter, P.Counter.Builder, Counter[F]](counter, labels, List.empty)
       }
 
+      def counterWithInitialLabels[A, B[_], D[_]](
+        name: String,
+        help: String,
+        labels: A)(implicit
+        magnet: LabelsMagnet[A, B, D]
+      ) = {
+        magnet.withInitialLabelValues { initialLabelValues =>
+          val counter = P.Counter.build()
+            .name(name)
+            .help(help)
 
-      def summary[A, B[_]](
+          apply[A, B, D, P.Counter.Child, P.Counter, P.Counter.Builder, Counter[F]](counter, labels, initialLabelValues)
+        }
+      }
+
+      def summary[A, B[_], D[_]](
         name: String,
         help: String,
         quantiles: Quantiles,
         labels: A)(implicit
-        magnet: LabelsMagnet[A, B]
+        magnet: LabelsMagnet[A, B, D]
```

Comment:
we cannot make backward incompatible changes…

Reply:
Do you think we have persons who implement B & D by self & and don't use predefined values?
Because I don't want to implement separate version of LabelsMagnet.

Reply:
I agree with you that this is the best way, but unfortunately I don't see a solution that would help us to avoid the boilerplate code in this case.

Reply:
I'm just trying yo say that you need to provide new `summary` method with new LabelsMagnet, so the old code could keep working - basically do changes in backward compatible way

Reply:
Good Day! Was created separate LabelsMagnet & part of changes were reverted on order to make code base backward compatible.